### DISCUSSION TITLE
roadrunner downloads index.php instead of serve it

### DISCUSSION THREAD

**Discussion by DrjavaB** [roadrunner-server/roadrunner#1831] [Q&A] [answered]
I crate a mini test project to serve my older php project via roadrunner but when I start rr server and open browser it downloads the index.php file and does not execute it at all.
here is my configs

.rr.yaml
```
version: "2023.3.8"
server:
  command: "php worker.php"

http:
  address: 127.0.0.1:8080
  middleware: [ "static" ]
  static:
    dir: "../shop"
    forbid: [ "" ]
    calculate_etag: true
    weak: true
    allow: [ ".php" ]
  pool:
    num_workers: 1
    debug: true

```

worker.php
```
<?php

use Nyholm\Psr7\Response;
use Spiral\Goridge;
use Spiral\RoadRunner;

ini_set('display_errors', 'stderr');
require __DIR__ . "/vendor/autoload.php";


$worker = RoadRunner\Worker::create();
$psr7 = new RoadRunner\Http\PSR7Worker(
    $worker,
    new \Nyholm\Psr7\Factory\Psr17Factory(),
    new \Nyholm\Psr7\Factory\Psr17Factory(),
    new \Nyholm\Psr7\Factory\Psr17Factory()
);

while ($req = $psr7->waitRequest()) {
    try {
        $resp = new Response();
        $resp->getBody()->write('Nothing to see here, move along...');
        $psr7->respond($resp);
    } catch (\Throwable $e) {
        $psr7->getWorker()->error((string)$e);
    }
}

```

index.php in shop directory

```
<?php

require dirname(__FILE__).'/config/config.inc.php';
Dispatcher::getInstance()->dispatch();
```

anyone can explain me how to serve php project with rr server?

**Comment by rustatian** [comment]
Hey @DrjavaB 👋 
Firstly, `..` in the path is not allowed. You should see a 403, IIRC status code. Next, you allowed `.php` extension. And since you have an `index.php` in the allowed folder with allowed extension, RR serves this file and the browser will download it as you requested.

**Reply by DrjavaB** [reply]
So how to serve it? what is the right way to launch general php project with rr server like laravel octane?

**Reply by rustatian** [reply]
Laravel Octane uses its own worker and Octane-specific commands. You may read their docs [here](https://laravel.com/docs/10.x/octane#roadrunner).

**Reply by DrjavaB** [reply]
as a result we can not serve pure php project via rr?

**Reply by rustatian** [reply]
I'm not sure, WDYM under the pure PHP project. Is Octane pure enough?

**Reply by DrjavaB** [reply]
yes! octane serve anything you write in laravel

**Reply by rustatian** [reply]
Then, please, read their docs, which I sent you in my previous response, specify RoadRunner as the server, and you are welcome to use it 😃
However, your RR experience will be limited by the `http` plugin, so if you want to use `Queues`, `gRPC`, real workflows (temporal.io), KV, WebSockets, etc - feel free to use our framework (SpiralFramework) or send PR (or feature request ticket) to the Octane.

**Reply by DrjavaB** [reply]
I realy love rr but our cto give me a task to speedup our php project and I select rr but I get confused how to serve index.php and project entry point without any minor change in it.

**Reply by rustatian** [reply, marked as answer]
Glad to hear that 👍 
With RR you have to create a worker. It is really simple since we're strictly following `PSR-7` modern approach. Here you may find a step-by-step instruction: [link](https://roadrunner.dev/docs/php-worker/current/en)
Without worker mode, you have to use `PHP-FPM`, `apache_mod`, `nginx`, etc. We're working on a modern solution, but it'll be presented next year.
